apps/admin-portal/src/components/roles/create-role-wizard/role-permisson.tsx

```diff
@@ -0,0 +1,207 @@
+/**
+ * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React, { FunctionComponent, ReactElement, useEffect, useState} from "react";
+import { getPermissionList, getPermissionsPerRole } from "../../../api";
+import { Permission } from "../../../models/permission";
+import { TreeView } from "@wso2is/react-components";
+import { Forms } from "@wso2is/forms";
+import { addPath } from "../role-utils";
+import { Segment, Button, Grid } from "semantic-ui-react";
+import _ from "lodash";
+
+/**
+ * Interface to capture permission list props
+ */
+interface PermissionListProp {
+    triggerSubmit?: boolean;
+    onSubmit?: (permissions: any) => void;
+    role?: string;
+}
+
+/**
+ * Component to create the permission tree structure from the give permission list.
+ * 
+ * @param props props containing event handlers for permission component
+ */
+export const PermissionList: FunctionComponent<PermissionListProp> = (props: PermissionListProp): ReactElement => {
+
+    const [ permissionTree, setPermissionTree ] = useState([]);
+    const [ selectedPermissions, setSelectedPermissions ] = useState([]);
+    const [ permissionsOfRole, setPermissionsOfRole ] = useState([]);
+    const [ permissionsLoading, setPermissionsLoading ] = useState(Boolean);
+    const [ collapseAll, setCollapseAll ] = useState(Boolean);
+
+    const {
+        triggerSubmit,
+        onSubmit,
+        role
+    } = props;
+
+    /**
+     * Util method to check permissions already in the selected role.
+     * 
+     * @param permList - Permissions which are available in the selected role
+     * @param nodes - Permmission Node List
+     * @param isParentChecked - Check whether parent is selected
+     */
+    const checkRolePerms = (permList: string[], nodes: any, isParentChecked: boolean): void  => {
```

Comment:
Shall we use qualified variable declarations e.g.  `checkRolePermissions` ? Noticed in other places as well

Reply:
fixed with commit 5fb1030